### ISSUE TITLE
Document state machine edge cases to help with troubleshooting incidents

### ISSUE THREAD

**Issue by anna-geller** [kestra-io/kestra#4840]
### Feature description

Create a decision tree of allowed state transitions + how to manage edge cases (to unblock users while cleaning up concurrency slots after user action like KILL or DELETE execution)

**Comment by anna-geller** [comment]
with features like Force Run, this is no longer needed 

![image](https://github.com/user-attachments/assets/d2b6b56c-a125-4f84-ac2b-df44d39a6852)
![image](https://github.com/user-attachments/assets/e8f09576-c83c-4b82-be14-2da860bab469)


Text:

Are you sure to force run execution 1pGEbPf9TqkpQBIvviWteP?

WARNING: force running an execution is not guaranteed and may create duplicate task executions.

The following transitions will be done:

CREATED tasks will be moved to the running state.
RUNNING tasks will be re-submitted.
QUEUED tasks will be un-queued
PAUSED tasks will be resumed.
